pageserver/src/walreceiver/walreceiver_connection.rs

```diff
@@ -409,3 +410,22 @@ async fn identify_system(client: &mut Client) -> anyhow::Result<IdentifySystem>
         Err(IdentifyError.into())
     }
 }
+
+fn ignore_expected_errors(pg_error: postgres::Error) -> anyhow::Result<postgres::Error> {
```

Comment:
I feel like this function should have a comment explaining _why_ it's OK to treat those errors as expected.
My (limited) understanding is:

```
/// We don't want to report connectivity problems as real errors towards connection manager because
/// 1. they happen frequently enough to make server logs hard to read and
/// 2. the connection manager can retry other safekeeper.
///
/// If this function returns `Ok(pg_error)`, it's such an error.
/// The caller should log it at info level and then report to connection manager that we're done handling this connection.
/// Connection manager will then handle reconnections.
///
/// If this function returns an `Err()`, the caller can bubble it up using `?`.
/// The connection manager will log the error at ERROR level.
```

Is tha tan adequate description of our error handling strategy in this module?

Reply:
More than adequate, copied it there, thank you.